spec/jobs/slack_notification_job_spec.rb

```diff
@@ -0,0 +1,63 @@
+require 'rails_helper'
+
+RSpec.describe SlackNotificationJob do
+  describe '#perform' do
+    it 'sends a Slack notification to this webhook if the URL is set' do
+      slack_request = stub_request(:post, 'https://example.com/webhook')
+        .to_return(status: 200, headers: {})
+
+      ClimateControl.modify STATE_CHANGE_SLACK_URL: 'https://example.com/webhook' do
```

Comment:
Further to https://github.com/DFE-Digital/find-teacher-training/pull/1099/files#r757390095 in the tests you can just a dummy key value pair to `config/settings/production.yml` i.e. 

`STATE_CHANGE_SLACK_URL: https://example.com/webhook`

If you then need to change the value then you can stub 
`allow(Settings).to receive(:state_change_slack_url).and_return('nil')`